upi/openstack/04_control-plane.yaml

```diff
@@ -43,38 +43,31 @@
     - os_networking_type == "Kuryr"
     - "os_cp_trunk_name|string not in control_plane_trunks.stdout"
 
-  - name: 'Get a token for creating the server group'
-    os_auth:
-    register: cloud
+  - name: 'List the Server groups'
+    command:
+      cmd: "openstack server group list -f json -c ID -c Name"
+    register: server_group_list
+
+  - name: 'Parse the Server group ID from existing'
+    set_fact:
+      server_group_id: "{{ (server_group_list.stdout | from_json | json_query(list_query) | first).ID }}"
```

Comment:
Nice!

Reply:
I particularly like the fact that in `server group list` you get `ID`s, whereas in `server group create` you get lowercase `id`s :man_facepalming: